Step 1: Wages with f-strings

In [1]:
name = "John"
hourly_wage = 15.65
hours = 35
weekly_pay = hourly_wage * hours
print(f"{name} worked {hours} hours and earned ${weekly_pay} this week.")


John worked 35 hours and earned $547.75 this week.


Step 2: Count the warm days

In [2]:
temps = [61, 65, 72, 58, 70, 74, 68]
high = 0
for t in temps:
    if t > 65:
        high = high + 1
print(high)

4


Step 3: Write a temperature converter

In [3]:
def F_to_C(f):
    c = (f-32)*5/9
    return round(c,1)
F_to_C(temps[0])

16.1

Step 4: A movie dictionary

In [4]:
movie = {"title": "Moneyball", "year": 2011, "rating": 7.6}
for key, value in movie.items():
    print(key, ":", value)

title : Moneyball
year : 2011
rating : 7.6


Step 5: Load and inspect chickwts.csv

In [5]:
import pandas as pd
df = pd.read_csv("chickwts.csv")
df.head()

,Unnamed: 0,weight,feed
0,1,179,horsebean
1,2,160,horsebean
2,3,136,horsebean
3,4,227,horsebean
4,5,217,horsebean


In [6]:
df.shape

(71, 3)

Step 6: Interpret describe()

In [7]:
df.describe()

,Unnamed: 0,weight
count,71.000000,71.000000
mean,36.000000,261.309859
std,20.639767,78.073700
min,1.000000,108.000000
25%,18.500000,204.500000
50%,36.000000,258.000000
75%,53.500000,323.500000
max,71.000000,423.000000


The most interesting thing about this dataset to me is the spread of the weights. The largest chick in the dataset is nearly four times as heavy as the lightest. This is also reflected by the relatively large standard deviation of the dataset (78.1) means the coefficient of variance (stdev/mean) is roughly 30%.